js/binance.js

```diff
@@ -271,7 +298,12 @@ module.exports = class binance extends Exchange {
     }
 
     async fetchMarkets (params = {}) {
-        const response = await this.publicGetExchangeInfo (params);
+        const marketType = this.options['defaultMarket'];
+        let method = 'publicGetExchangeInfo';
+        if (marketType === 'futures') {
+            method = 'fapiPublicGetExchangeInfo';
+        }
```

Comment:
How do you think of changing the code`const symbol = base + '/' + quote;` which written in `fetchMarkets()`.
The market symbols of `binance futures` should be like `USDT-BTC-SWAP`, `USDT-ETH-SWAP` which could match the pattern of ccxt `swap` symbols. @kroitor @StanleyDing

Reply:
@lyn716 i'm currently checking if we can load both types of markets and have separated symbols for futures. On one hand, that may be convenient. However, on the other hand, that may raise questions from users who are not very familiar with CCXT but have previous experience with Binance API. This is still an open question to me, but thx for your considerations. Doing all I can to merge it within a few hours, hopefully, today.

Reply:
Thanks for your mentioned above. It is something I had not considered from new users' scope. 

Our team also built unified-exchange stack which based on ccxt, `market` and `symbol` are both serious objects in our scope. If the market is type of `SWAP`, we will recognize them by two things of ccxt: `market["symbol"]` and `market["type"]`.

But this PR lost them both. So we have to override it with new symbol converting function. It is tiny problem we have.

Other content of this PR are helpful enough. Very thanks! @StanleyDing @kroitor

Reply:
@lyn716 

> we will recognize them by two things of ccxt: `market["symbol"]` and `market["type"]`

We will definitely have the `market['type']` property for all binance markets, so you will have at least one way of detecting them. I'm not completely sure about the second option with `SWAP/FUTURE` in the market symbol, but will figure this out shortly. We will have at least one way of differentiating them – that's for sure.

Reply:
One thing I think the market data is missing is the lack of `settleCurrency`. In binance futures, `BTC/USDT` contract is settled in USDT while Bitmex is settled in BTC. Currently we can't get this information from the market data.

Reply:
@StanleyDing we will definitely look into unifying more market properties, including properties specific to different types of markets... Need a bit more of futures implementations to come up with a meaningful abstraction. Apart from the `settleCurrency, i'm sure, there are many properties a user might want in his logic.